modules/apm/src/test/java/org/elasticsearch/telemetry/apm/MeterRegistryConcurrencyTests.java

```diff
@@ -106,12 +105,14 @@ public void testLockingWhenRegistering() throws Exception {
         // assert that a thread is waiting for a lock during long-running registration
         assertBusy(() -> assertThat(setProviderThread.getState(), equalTo(Thread.State.WAITING)));
         // assert that the old lockingMeter is still in place
-        assertBusy(() -> assertThat(meterRegistrar.getMeter(), sameInstance(lockingMeter)));
+        assertThat(meterRegistrar.getMeter(), sameInstance(lockingMeter));
 
         // finish long-running registration
         registerLatch.countDown();
+        // wait for everything to quiesce, registerLatch.countDown() doesn't ensure lock has been released
+        assertBusy(setProviderThread::join);
```

Comment:
join is blocking, so assertBusy shouldn't be necessary?

Reply:
done.